src/io/DumpH5MDParallel.cpp

```diff
@@ -0,0 +1,322 @@
+    std::vector<hsize_t> stride(globalDims.size(), 1);
+    std::vector<hsize_t> count(globalDims.size(), 1);
+    for (auto i = 0; i < globalDims.size(); ++i) {
+        CHECK_LESS_EQUAL(localDims[i] + offset[i], globalDims[i], "i = " << i);
+    }
+    auto dstSpace = CHECK_HDF5(H5Dget_space(dataset));
+    CHECK_HDF5(H5Sselect_hyperslab(dstSpace, H5S_SELECT_SET, offset.data(), stride.data(),
+                                   count.data(), localDims.data()));
+
+    std::vector<hsize_t> localOffset(globalDims.size(), 0);
+    auto srcSpace = CHECK_HDF5(H5Screate_simple(localDims.size(), localDims.data(), NULL));
+    CHECK_HDF5(H5Sselect_hyperslab(srcSpace, H5S_SELECT_SET, localOffset.data(), stride.data(),
+                                   count.data(), localDims.data()));
+
+    auto datawrite_id = CHECK_HDF5(H5Pcreate(H5P_DATASET_XFER));
+    CHECK_HDF5(H5Pset_dxpl_mpio(datawrite_id, H5FD_MPIO_COLLECTIVE));
+    CHECK_HDF5(H5Dwrite(dataset, typeToHDF5<T>(), srcSpace, dstSpace, datawrite_id, data.data()));
+
+    CHECK_HDF5(H5Pclose(datawrite_id));
+    CHECK_HDF5(H5Sclose(dstSpace));
+    CHECK_HDF5(H5Sclose(srcSpace));
+    CHECK_HDF5(H5Dclose(dataset));
+    CHECK_HDF5(H5Sclose(dataspace));
+}
+
+void DumpH5MDParallel::writeHeader(hid_t fileId) {
+    auto group1 = H5Gcreate(fileId, "/h5md", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    auto group2 = H5Gcreate(fileId, "/h5md/author", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    auto group3 = H5Gcreate(fileId, "/h5md/creator", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    CHECK_HDF5(H5Gclose(group1));
+    CHECK_HDF5(H5Gclose(group2));
+    CHECK_HDF5(H5Gclose(group3));
+
+    std::vector<int> data = {2, 0};
+    CHECK_HDF5(H5LTset_attribute_int(fileId, "/h5md", "version", data.data(), data.size()));
+
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/author", "name", author.c_str()));
+
+    std::string software = "espressopp";
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/creator", "name", software.c_str()));
+
+    std::string version = Version().info();
+    CHECK_HDF5(H5LTset_attribute_string(fileId, "/h5md/creator", "version", version.c_str()));
+}
+
+void DumpH5MDParallel::writeBox(hid_t fileId) {
+    std::string groupName = "/particles/atoms/box";
+    auto group = CHECK_HDF5(H5Gcreate(fileId, groupName.c_str(), H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+    std::vector<int> dims = {3};
+    CHECK_HDF5(H5LTset_attribute_int(fileId, groupName.c_str(), "dimension", dims.data(), dims.size()));
+
+    auto boundaryType = H5Tcopy(H5T_C_S1);
+    CHECK_HDF5( H5Tset_size(boundaryType, 8) );
+    CHECK_HDF5( H5Tset_strpad( boundaryType, H5T_STR_NULLPAD ));
+    std::vector<hsize_t> boundaryDims = {3};
+    auto space = H5Screate_simple(boundaryDims.size(), boundaryDims.data(), nullptr);
+    auto att = H5Acreate(group, "boundary", boundaryType, space, H5P_DEFAULT, H5P_DEFAULT);
+    CHECK_HDF5(H5Awrite(att, boundaryType, "periodicperiodicperiodic"));
```

Comment:
Why do you have to  put here `periodiperiodicperiodic` ?

Reply:
Judging from
https://github.com/espressopp/espressopp/blob/a967dada2b0fda6d390b89abcdc1f1541068d823/src/io/DumpH5MD.py#L216-L227
this seems to be the only supported case.

Reply:
So this `periodicperiodicperiodic` will be written as three element array?

Reply:
yes